### PR TITLE
feat(ui_auth): add display name screen when user register

### PR DESCRIPTION
**Pull request by YassinNouh21**
## Description

This PR enhances the ```SignInScreen``` widget by adding a new feature. When the `displayName` property of the user is found to be `null`, the app will now navigate the user to a screen titled "What is your name?" where they can input or edit their display name. This feature provides a more user-friendly.

## Related Issues

This is based on [Discussed in 69](https://github.com/firebase/FirebaseUI-Flutter/discussions/69) and [Issue 76 ](https://github.com/firebase/FirebaseUI-Flutter/issues/76)for enhancement 

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for _all_ changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] All unit tests pass (`melos run test:unit:all` doesn't fail).
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is _not_ a breaking change.

[issue database]: https://github.com/firebase/FirebaseUI-Flutter/issues
[Contributor Guide]: https://github.com/firebase/FirebaseUI-Flutter/blob/main/docs/contributing.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by YassinNouh21 — COMMENTED** [review]
Thank you for your feedback, I just add in the screen folder new screen `DisplayNameScreen` with its text controller `EditableSignUserDisplayName` and added action for navigating when user finishes typing his name

File: packages/firebase_ui_auth/lib/src/actions.dart [hunk 1]
```diff
@@ -2,8 +2,8 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'package:flutter/material.dart';
 import 'package:firebase_ui_auth/firebase_ui_auth.dart';
+import 'package:flutter/material.dart';
```
Comment (lesnitsky) [inline packages/firebase_ui_auth/lib/src/actions.dart:6]: This change is irrelevant to the PR's main goal.

File: packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+import 'package:firebase_ui_localizations/firebase_ui_localizations.dart';
+import 'package:firebase_ui_shared/firebase_ui_shared.dart';
+import 'package:flutter/material.dart' hide Title;
+import 'package:flutter/scheduler.dart';
+
+import '../widgets/editable_sign_user_display_name.dart';
+import 'internal/responsive_page.dart';
+
+/// An action that is being called when sign up was successfully.
+class DisplayNameAction extends FirebaseUIAction {
```
Comment (lesnitsky) [inline packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart:16]: I'm not sure what this action does and the doc comment is confusing. Why is it called after sign up but called `DisplayNameAction`?

File: packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  bool isLoading = false;
+
+  @override
+  void initState() {
+    _ambiguate(SchedulerBinding.instance)!
+        .addPostFrameCallback(_sendEmailVerification);
+    super.initState();
+  }
+
+  void _sendEmailVerification(_) {
```
Comment (lesnitsky) [inline packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart:143]: Why this screen has anything to do with the email verification?

File: packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Align(
+          child: Text(
+            l.name,
+            style: Theme.of(context).textTheme.headline6,
+          ),
+        ),
+        const SizedBox(height: 32),
+        EditableSignUserDisplayName(auth: auth),
```
Comment (lesnitsky) [inline packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart:176]: there is an existing widget: [`EditableUserDisplayName`](https://pub.dev/documentation/firebase_ui_auth/latest/firebase_ui_auth/EditableUserDisplayName-class.html)

File: packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+  Widget build(BuildContext context) {
+    final l = FirebaseUILocalizations.labelsOf(context);
+
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Align(
+          child: Text(
+            l.name,
```
Comment (lesnitsky) [inline packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart:171]: this label is used in a different context and might change, this screen needs its own label (`editableDisplayNameTitle`). Currently adding a new label is a bit tedious, but there are a few improvements to this process in #116, so it's worth waiting for it to be merged first.

File: packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            style: Theme.of(context).textTheme.headline6,
+          ),
+        ),
+        const SizedBox(height: 32),
+        EditableSignUserDisplayName(auth: auth),
+      ],
```
Comment (lesnitsky) [inline packages/firebase_ui_auth/lib/src/screens/display_name_screen.dart:177]: This screen needs `continue` and `go back` buttons. With current implementation, if `DisplayNameChanged` action is not provided – the user will be stuck on this screen. I'm doubting that we need this action at all. `continue` and `go back` buttons should be enough.

File: packages/firebase_ui_auth/lib/src/widgets/editable_sign_user_display_name.dart [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+import 'package:firebase_auth/firebase_auth.dart' show FirebaseAuth;
+import 'package:firebase_ui_localizations/firebase_ui_localizations.dart';
+import 'package:firebase_ui_shared/firebase_ui_shared.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+import '../../firebase_ui_auth.dart';
+import 'internal/subtitle.dart';
+
+class EditableSignUserDisplayName extends StatefulWidget {
```
Comment (lesnitsky) [inline packages/firebase_ui_auth/lib/src/widgets/editable_sign_user_display_name.dart:14]: https://pub.dev/documentation/firebase_ui_auth/latest/firebase_ui_auth/EditableUserDisplayName-class.html

**Review by lesnitsky — CHANGES_REQUESTED** [review]

**Comment by lesnitsky** [comment]
@YassinNouh21 would you be willing to address my feedback?

**Comment by lesnitsky** [comment]
@YassinNouh21 any updates on this PR?

**Comment by YassinNouh21** [comment]
> @YassinNouh21 any updates on this PR?

Sorry I been busy this week I will finish it by the weekend

**Comment by lesnitsky** [comment]
@YassinNouh21 any updates on this?

**Comment by bakua** [comment]
Shame :/

**Pull request closed without merging**